app/src/main/java/com/example/candyshop/Navigation.kt

```diff
@@ -0,0 +1,49 @@
+package com.example.candyshop
+
+import androidx.compose.animation.AnimatedContentTransitionScope
+import androidx.compose.animation.core.tween
+import androidx.compose.runtime.Composable
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+
+@Composable
+fun Navigation() {
+    val navController = rememberNavController()
+    val time = 500
+    NavHost(
+        navController = navController,
+        startDestination = Screen.MainScreen.route,
+        enterTransition = {
+            slideIntoContainer(AnimatedContentTransitionScope.SlideDirection.Left, tween(time))
+        },
+        exitTransition = {
+            slideOutOfContainer(AnimatedContentTransitionScope.SlideDirection.Left,tween(time))
+        },
+        popEnterTransition = {
+            slideIntoContainer(AnimatedContentTransitionScope.SlideDirection.Right,tween(time))
+        },
+        popExitTransition = {
+            slideOutOfContainer(AnimatedContentTransitionScope.SlideDirection.Right,tween(time))
+        }
+    ) {
+        composable(route = Screen.MainScreen.route) {
+            CandyShopMain(navController = navController)
+        }
+        composable(
+            route = Screen.DetailsScreen.route + "/{name}",
+            arguments = listOf(
+                navArgument("name") {
```

Comment:
relying on naming when we are passing data between list and details view can be quite dangerous, maybe we could create some unique id in our structure and rely on it?

Reply:
I tried using UUID for the id, but new one is generated every time the app runs. It worked fine in the emulator, but because it changes every time I couldn't make a preview using UUID, so instead I just hardcoded an int id.